server/Arcadia.Assistant.Web/Controllers/HealthController.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Arcadia.Assistant.Web.Controllers
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Health;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Models;
+
+    public class HealthController : Controller
+    {
+        private readonly IHealthService healthService;
+
+        public HealthController(IHealthService healthService)
+        {
+            this.healthService = healthService;
+        }
+
+        [Route("/api/ping/health")]
+        [HttpGet]
+        [ProducesResponseType(typeof(ApplicationHealthModel), StatusCodes.Status200OK)]
+        public async Task<IActionResult> GetApplicationHealth(CancellationToken cancellationToken)
+        {
+            var tasks = new[]
+            {
+                this.healthService.GetIsServerAlive(cancellationToken),
```

Comment:
what should be happening is that
1. web side requests a health check (with a default timeout * 2)
    -> not given response - return an error.
2. health actor on server side (has CSP implementation)
    -> asks vacationsactor and databaseactor about their status (with defaulttimeout)
    -> return a dictionary.

Reply:
Why "* 2"? Why not 3 or 5?

Reply:
1. Web controller makes a request to health service that asks health actor about current server health state.
If actor is not responding - health service returns dictionary with one entry: { server: false }.
If actor is returning result - health service adds entry { server: true } and returns it.
2. On server side health actor makes checks for 1C and database availability by itself and returns dictionary of health state entries. The only remark is that there is no timeout setting in CSP project, it is web specific. So, if either database or 1C check takes a bit more time - server in whole will be considered as not alive.

Is this what you wanted?

Reply:
2. Fixed to get status from existing actor